python/src/kestra.py

```diff
@@ -37,6 +44,222 @@ def timer(name, duration, tags=None):
+        if response.status_code == 401:
+            raise Exception(
+                "Authentication required but not provided. Please set the username and password."
+            )
+
+        response.raise_for_status()
+        return response
+
+    def check_status(self, execution_id: str) -> requests.Response:
+        """
+        Check the status of the execution
+        :param execution_id: the ID of the execution
+        :return: the response from the server
+        """
+        return self._make_request(
+            "get",
+            self.hostname
+            + Flow.API_ENDPOINT_EXECUTION_STATUS.replace(
+                "PARAM_EXECUTION_ID", execution_id
+            ),
+        )
+
+    def get_logs(self, execution_id: str) -> requests.Response:
+        """
+        Get the execution logs
+        :param execution_id: the ID of the execution
+        :return: the response from the server
+        """
+        return self._make_request(
+            "get",
+            self.hostname
+            + Flow.API_ENDPOINT_EXECUTION_LOG.replace(
+                "PARAM_EXECUTION_ID", execution_id
+            ),
+        )
+
+    def execute(
+        self,
+        namespace: str,
+        flow: str,
+        inputs: dict = None,
+    ) -> namedtuple:
+        if inputs is None:
+            inputs = {}
+
+        logging.debug(
+            "Starting a flow %s in the namespace %s with parameters %s",
+            flow,
+            namespace,
+            str(inputs),
+        )
+        result = namedtuple("FlowExecution", ["status", "log", "error"])
+        url_default = self.hostname + Flow.API_ENDPOINT_EXECUTION_CREATE.replace(
+            "PARAM_FLOW_ID", namespace + "/" + flow
+        )
+        if self.labels_from_inputs and len(inputs) > 0:
+            if inputs.get("files") is not None:
+                raise Exception(
+                    "Labels from inputs is not supported with inputs of type FILE"
+                )
+            labels = "?"
+            for key in inputs:
+                key_ = urllib.parse.quote(key)
+                value_ = urllib.parse.quote(inputs[key])
+                labels = f"{labels}labels={key_}:{value_}"
+
+            url = url_default + labels
+            response = self._make_request("post", url, files=inputs).json()
+        elif len(inputs) > 0:
+            response = self._make_request("post", url_default, files=inputs).json()
+        else:
+            response = self._make_request("post", url_default).json()
+
+        if "id" not in response:
+            raise Exception("Starting execution failed: " + str(response))
+
+        execution_id = response["id"]
+        logging.info(
+            "Successfully triggered the execution: %s/ui/executions/%s/%s/%s",
+            self.hostname,
+            namespace,
+            flow,
+            execution_id,
+        )
+        if self.wait_for_completion:
+            finished = False
+            while not finished:
```

Comment:
Handle KILLED status
Also, what happen if the execution is deleting during the execution ?

Reply:
well spotted, I'll add KILLED handling

for deleted execution, I believe checking for the status of that execution ID will return an error, which seems expected, WYT?

Reply:
added now

Reply:
@Skraye LMK if this is good to go now